src/cpp/src/sampler.cpp

```diff
@@ -658,14 +755,29 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
         // NOTE: it should be before 'get_num_scheduled_tokens' is used
         // update internal state of sequence group to reset scheduler tokens and update currently processed ones
         sequence_group->finish_iteration();
+        // decrease sequence_group context in case of candidates generated by draft_model were not accepted by main_model
+        if (decrease_context_len_per_seq_group) {
+            const auto num_processed_tokens = sequence_group->get_num_processed_tokens();
+            OPENVINO_ASSERT(num_processed_tokens >= decrease_context_len_per_seq_group);
+            OPENVINO_ASSERT(sequence_group->get_context_len() >= decrease_context_len_per_seq_group);
+            sequence_group->update_processed_tokens_num(num_processed_tokens - decrease_context_len_per_seq_group);
+        }
 
         // accumulate a number of processed tokens
-        currently_processed_tokens += padded_amount_of_processed_tokens * num_running_sequences;
+        currently_processed_tokens += (padded_amount_of_processed_tokens - decrease_context_len_per_seq_group) * num_running_sequences;
```

Comment:
why do we need to subtract `decrease_context_len_per_seq_group` here? 
We are iterating over already computed logits tensor and looks like validation passed / failed does not make sense here

Reply:
to get a correct offset here
```
const void * sequence_group_logits_data = logits_data + vocab_size * currently_processed_tokens;
```

Reply:
agreed to fix separately